### PR TITLE
Disable native varargs for Windows arm

### LINKED ISSUE #35754: ArgIteratorTests.ArgIterator_Throws_PlatformNotSupportedException test fails on Windows ARM
```
System.Tests.ArgIteratorTests.ArgIterator_Throws_PlatformNotSupportedException [FAIL]
      Assert.Throws() Failure
      Expected: typeof(System.PlatformNotSupportedException)
      Actual:   typeof(System.ArgumentException): Handle is not initialized.
      ---- System.ArgumentException : Handle is not initialized.
      Stack Trace:
           at System.ArgIterator..ctor(IntPtr arglist)
        /_/src/coreclr/src/System.Private.CoreLib/src/System/ArgIterator.cs(35,0): at System.ArgIterator..ctor(RuntimeArgumentHandle arglist)
        /_/src/libraries/System.Runtime/tests/System/ArgIteratorTests.cs(93,0): at System.Tests.ArgIteratorTests.<>c.<ArgIterator_Throws_PlatformNotSupportedException>b__6_0()
        ----- Inner Stack Trace -----
           at System.ArgIterator..ctor(IntPtr arglist)
        /_/src/coreclr/src/System.Private.CoreLib/src/System/ArgIterator.cs(35,0): at System.ArgIterator..ctor(RuntimeArgumentHandle arglist)
        /_/src/libraries/System.Runtime/tests/System/ArgIteratorTests.cs(93,0): at System.Tests.ArgIteratorTests.<>c.<ArgIterator_Throws_PlatformNotSupportedException>b__6_0()
```

Var args is supposed to throw not supported exception in Windows ARM.

For more info, see: https://github.com/dotnet/runtime/pull/35690#issuecomment-622597724

cc: @jkotas @jashook

### PR DESCRIPTION
**Pull request by kunalspathak**
Disable native varargs for Windows arm so `ArgsIterator` throws `PlatformNotSupported` exception. Also re-enable the test that was disabled in https://github.com/dotnet/runtime/pull/35690.

Fixes: https://github.com/dotnet/runtime/issues/35754

### CODE REVIEW

File: src/coreclr/src/jit/target.h [hunk 1]
```diff
@@ -6,7 +6,7 @@
 #ifndef TARGET_H_
 #define TARGET_H_
 
-#if defined(TARGET_UNIX)
+#if defined(TARGET_UNIX) || defined(TARGET_ARM)
```
Comment (jkotas) [inline src/coreclr/src/jit/target.h:9]: Flip the logic to: `#if defined(TARGET_WINDOWS) && !defined(TARGET_ARM)` to make it look like the condition in other places.

File: src/coreclr/src/jit/target.h [hunk 2]
```diff
@@ -6,10 +6,11 @@
 #ifndef TARGET_H_
 #define TARGET_H_
 
-#if defined(TARGET_UNIX)
-#define FEATURE_VARARG 0
-#else
+ // Native Varargs are not supported on Unix (all architectures) and Windows ARM
+#if (TARGET_WINDOWS && !TARGET_ARM)
```
Comment (jkotas) [inline src/coreclr/src/jit/target.h:10]: Nit: Everything else in this file uses `defined(TARGET_ARM)`, etc.

Comment (kunalspathak) [reply src/coreclr/src/jit/target.h:10]: Ah, yes. Sorry, you mentioned it already but didn't realized `defined` part. Will fix it.

**Review by jkotas — APPROVED** [review]

**Comment by kunalspathak** [comment]
Manually triggered runtime pipeline and leg "Libraries Test Run release coreclr Windows_NT arm Release" passed.
Merging because the last commit is formatting.

**Pull request merged**